### ISSUE TITLE
Should breeds inherit defaults?

### ISSUE THREAD

**Issue by bennlich** [backspaces/agentscript#83]
I thought they used to, and that something changed, because now they don't. But I'm not positive they ever did. Do you know what was intended? E.g.

```
this.agentBreeds("cakes fish");
this.agents.setDefault("size", 5);
```

Should cakes and fish have default sizes of 5?

**Comment by backspaces** [comment]
Nope. AS clones the Agent class for each breed before defaults get a chance
to be made.

Good point tho.  I wonder if NL does this, it certainly makes sense.

If we convert from cloning classes for defaults/multimodel to shimming in a
defaults layer, we could implement this behavior.

**Comment by bennlich** [comment]
K. Does order matter? Like if you set defaults and then create breeds?
Ideally order would not matter, I think.

**Comment by backspaces** [comment]
>
>
> K. Does order matter? Like if you set defaults and then create breeds?
> Ideally order would not matter, I think.
>
>
​Not sure I understand.
But I suspect you mean​

​"can I set defaults on agents, then on ​fish" .. will they inherit from
each other?  They would with our current clone stunt, I think.  But because
the breeds are created within the constructor, there's no way for them to
do so .. the breeds get created with the vanilla Patch/Agent/Link classes
before they can have any defaults applied.

So basically I think defaults are independent of Foo and a Foo breed.  I
think that is the way I'd expect it to work, given that Patch/Agent/Link
are not really being subclassed by breeds, but just copied but with
different "own" variables.

I think if we enforced the "own" variables ( i.e. agents.own("age weight
height") ) it would make things much clearer .. and make it easier to
convert between breeds .. and catch spelling errors etc.  We could do it by
making the individual agents non-extensible maybe.

**Comment by backspaces** [comment]
Hmm.. looking at the NL docs, I could only find set-default-shape.  I
thought you could default color, size etc but apparently not.

I'm having difficulty integrating the new typedColor into AS, partly due to
how defaults currently work.  If I go back to something like our earlier
approach: adding an additional object shimmed into the class for the
defaults, then we might be able to have the sorta expected behavior you
mentioned.
